pkg/v2/service.go

```diff
@@ -565,8 +565,37 @@ func (s *service) Shutdown(ctx context.Context, r *taskAPI.ShutdownRequest) (*pt
 }
 
 func (s *service) Stats(ctx context.Context, r *taskAPI.StatsRequest) (*taskAPI.StatsResponse, error) {
-	// TODO(random-liu): Use `runsc events --stats`.
-	data, err := typeurl.MarshalAny(&cgroups.Metrics{})
+	path, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+	ns, err := namespaces.NamespaceRequired(ctx)
+	if err != nil {
+		return nil, err
+	}
+	runtime, err := s.readRuntime(path)
+	if err != nil {
+		return nil, err
+	}
+	rs := proc.NewRunsc(proc.RunscRoot, path, ns, runtime, nil)
```

Comment:
At this point, a runsc process should already exist:
```
	p, err := s.getProcess(r.ExecID)
	if err != nil {
		return nil, err
	}
```

Reply:
Yes, the process should already exist, but the `Stats` method is only exposed on the container, not on [`rproc.Process`](https://pkg.go.dev/github.com/containerd/containerd/runtime/proc?tab=doc#Process).  Unless I'm misinterpreting what you're meaning by this comment?

Also, as a side note, gvisor appears to [ignore](https://github.com/google/gvisor/blob/277a0d5a1fbe8272d4729c01ee4c6e374d047ebc/runsc/boot/events.go#L62) the input anyway and return container-wide metrics in all cases?

Reply:
Never mind. I thought for a second that getProcess would return `*runsc.Runsc`.

Yes, gVisor doesn't track per-container stats at the moment (https://github.com/google/gvisor/issues/172).